sphinx/source/docs/quickstart.rst

```diff
@@ -6,374 +6,585 @@ Quickstart
-Here's how the code for this use case looks like using bokeh.charts:
+Plots are a central concept in Bokeh. They are containers that hold all the
+various objects (renderers, guides, data, and tools) that comprise the final
+visualization that is presented to users. The |bokeh.plotting| interface
+provides a |Figure| class to help with assembling all the necessary objects,
+and a convenience function |figure| for creating |Figure| objects.
 
-.. bokeh-plot::
-   :source-position: above
+Glyphs
+------
 
-   import numpy as np
-   from bokeh.plotting import figure, output_file, show
+Glyphs are the basic visual marks that Bokeh can display. At the lowest level,
+there are **glyph objects**, such as |Line|. If you are using the low-level
+|bokeh.models| interface, it is your responsibility to create and coordinate
+all the various Bokeh objects, including glyph objects and their data sources.
+To make life easier, the |bokeh.plotting| interface exposes higher level
+**glyph methods** such as the |Figure.line| method used in the first example.
+The second example also adds in calls to |Figure.circle| to display circle
+and line glyphs together on the same plot. Besides lines and circles, Bokeh
+makes many additional |glyphs| and |markers| available.
+
+The visual appearance a glyphs is tied directly to that data values that are
+associated with the glyph's various attributes. In the example above we see
+that positional attributes like `x` and `y` can be set to vectors of data.
+But glyphs also have some combination of |line_props|, |fill_props|, and
+|text_props| to control their appearance. All of these attributes can be set
+with "vectorized" values as well. We will show examples of this below.
+
+Guides and Annotations
+----------------------
+
+Bokeh plots can also have other visual components that aid presentation or
+help the user make comparisons. These fall into two categories. **Guides**
+are visual aids that help users judge distances, angles, etc. These include
+grid lines or bands, axes (such as linear, log, or datetime) that may have
+ticks and tick labels as well. **Annotations** are visual aids that label or
+name parts of the plot. These include titles, legends, etc.
 
-   # prepare data
-   mu, sigma = 0, 0.5
-   measured = np.random.normal(mu, sigma, 1000)
-   hist, edges = np.histogram(measured, density=True, bins=50)
-   x = np.linspace(-2, 2, 1000)
+Ranges
+------
 
-   # output to static HTML file
-   output_file('histogram.html')
+Ranges describe the data-space bounds of a plot. By default, plots generated
+with the |bokeh.plotting| interface come configured with
+:class:`DataRange1d <bokeh.models.ranges.DataRange1d>` objects that try to
+automatically set the plot bounds to encompass all the available data.
+But it is possible to supply explicit
+:class:`Range1d <bokeh.models.ranges.Range1d>` objects for fixed bounds.
+As a convenience these can also typically be spelled as 2-tuples or lists::
```

Comment:
Maybe it'd be a nice addition to have a screenshot of a figure where we point all the elements listed above (thinking more about beginner users)?